lib/puppet/provider/group/windows_adsi.rb

```diff
@@ -22,9 +22,6 @@ def members_insync?(current, should)
 
     # Cannot use munge of the group property to canonicalize @should
     # since the default array_matching comparison is not commutative
-    should_empty = should.nil? or should.empty?
-
-    return false if current.empty? != should_empty
```

Comment:
Yanked. If we put this back for any reason, it also has a bug in should_empty set because assignment to logical boolean.

So it would need to come back as 
`should_empty = should.nil? || should.empty?`